Fragment debug_id u16 only exists in debug, prod will format mem address

mitchhentges opened the pull request:
<!-- Please describe your changes on the following line: -->

Each fragment has a `u16` `debug_id` in debug mode, but no `debug_id` in production to save memory. To format a debug id in production, the address of the empty `debug_id` is displayed.

---

<!-- Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `__` with appropriate data: -->
- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #87 (github issue number if applicable).

<!-- Either: -->
- [X] These changes do not require tests because it looks like it's not possible to mock out `cfg` options in `#[test]`s

<!-- Pull requests that do not address these steps are welcome, but they will require additional verification as part of the review process. -->

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/11442)

<!-- Reviewable:end -->

Commit:
Fragment debug_id u16 only exists in debug, prod will format mem address

highfive commented:
<img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20> **Warning** <img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20>

* These commits modify layout code, but no tests are modified. Please consider adding a test!

mitchhentges commented:
CI is mad right out-of-the-gate :(

highfive commented:
New code was committed to pull request.

mitchhentges commented:
Merged master to make CI go again (Appveyor had failed immediately for the first commit)

jdm commented:
If appveyor shows a build error unrelated to your changes, it's not a big deal.

KiChjang commented on components/layout/fragment.rs:
```diff
@@ -2780,3 +2776,57 @@ pub struct SpeculatedInlineContentEdgeOffsets {
+
+#[cfg(debug_assertions)]
+#[derive(Clone)]
+struct DebugId {
```
This could just be `struct DebugId(u16)`.

highfive commented:
New code was committed to pull request.

mitchhentges commented:
I also made the `not(debug_assertions)` one a "unit-like" struct

KiChjang commented:
@bors-servo r+

Thanks!

bors-servo commented:
:pushpin: Commit 14a90d0 has been approved by `KiChjang`

<!-- @bors-servo r=KiChjang 14a90d0e5d520566e66529d3ff4282ac021423bb -->

bors-servo commented:
:hourglass: Testing commit 14a90d0 with merge 170a21e...

bors-servo commented:
:broken_heart: Test failed - [linux-rel](http://build.servo.org/builders/linux-rel/builds/1359)

KiChjang commented:
Fails compiling release:
```
/home/servo/buildbot/slave/linux-rel/build/components/layout/fragment.rs:2819:31: 2819:37 error: invalid reference to argument `0` (no arguments given)
/home/servo/buildbot/slave/linux-rel/build/components/layout/fragment.rs:2819         e.emit_string(format!("{:p}"), &self)
                                                                                                            ^~~~~~
<std macros>:2:28: 2:59 note: in this expansion of format_args!
/home/servo/buildbot/slave/linux-rel/build/components/layout/fragment.rs:2819:23: 2819:38 note: in this expansion of format! (defined in <std macros>)
error: aborting due to previous error
error: Could not compile `layout`.
```

highfive commented:
New code was committed to pull request.

mitchhentges commented:
Ah, you do `./mach build --release` to build without `debug_assertions` to test the other compile path, cool.
Fixed, and I also added some `#[cfg(not(debug_assertions))]` to try and not create `DEBUG_ID_COUNTER`, ` generate_unique_debug_id()` and the import for `layout_debug`, but I'm still getting warnings in the console. Maybe I should ignore the "dead code" warning?

```
/home/mitch/dev/servo/components/layout/layout_debug.rs:23:1: 23:58 warning: static item is never used: `DEBUG_ID_COUNTER`, #[warn(dead_code)] on by default
/home/mitch/dev/servo/components/layout/layout_debug.rs:23 static DEBUG_ID_COUNTER: AtomicUsize = ATOMIC_USIZE_INIT;
                                                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/mitch/dev/servo/components/layout/layout_debug.rs:101:1: 103:2 warning: function is never used: `generate_unique_debug_id`, #[warn(dead_code)] on by default
/home/mitch/dev/servo/components/layout/layout_debug.rs:101 pub fn generate_unique_debug_id() -> u16 {
                                                            ^
/home/mitch/dev/servo/components/layout/fragment.rs:53:5: 53:17 warning: unused import, #[warn(unused_imports)] on by default
/home/mitch/dev/servo/components/layout/fragment.rs:53 use layout_debug;

mitchhentges commented:
Never mind, found my mistake

highfive commented:
New code was committed to pull request.

mitchhentges commented:
I realize my mistake, and now the warnings are resolved

KiChjang commented:
Fails tidy:
```
./components/layout/fragment.rs:52: encountered whitespace following a use statement

./components/layout/layout_debug.rs:19: encountered whitespace following a use statement
```

mitchhentges commented:
Gah, thought I could multitask with the end of the semester coming up! I'll
finish this pull request this weekend when I've got more time.

Thanks for your patience :)

Fails tidy:

./components/layout/fragment.rs:52: encountered whitespace following a
use statement

./components/layout/layout_debug.rs:19: encountered whitespace
following a use statement

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/servo/servo/pull/11442#issuecomment-223034255>, or mute
the thread
<https://github.com/notifications/unsubscribe/AHbJIS4FSIhcEZ4G9dHzLahx0Gyq5O39ks5qHahfgaJpZM4InusO>
.

highfive commented:
New code was committed to pull request.

mitchhentges commented:
Okay. I think I've got this covered. I've:
* Run `./mach test-tidy`
* Run `./mach build -d`
* Run `./mach build --release`

I don't know what else I can do other than run all tests locally, which will take a little too long. So, I'm pretty confident that it should pass CI now :smile:

KiChjang commented:
Fails tidy:
```
./components/layout/fragment.rs:52: encountered whitespace following a use statement

./components/layout/layout_debug.rs:19: encountered whitespace following a use statement
```

mitchhentges commented:
I must've made a mistake on my git add, agh. Handling this outside my ide (for amends) is throwing me off

highfive commented:
New code was committed to pull request.

KiChjang commented:
@bors-servo r+

bors-servo commented:
:pushpin: Commit 43396c0 has been approved by `KiChjang`

<!-- @bors-servo r=KiChjang 43396c027dcbc357f0fef674e55f9413f56f7535 -->

bors-servo commented:
:hourglass: Testing commit 43396c0 with merge a149d75...

bors-servo commented:
:broken_heart: Test failed - [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/1417)

mitchhentges commented:
I don't think that the failing build is due to my changes

KiChjang commented:
@bors-servo retry
* The annoying git

bors-servo commented:
:hourglass: Testing commit 43396c0 with merge 5002dff...

bors-servo commented:
:sunny: Test successful - [android](http://build.servo.org/builders/android/builds/1298), [arm32](http://build.servo.org/builders/arm32/builds/1385), [arm64](http://build.servo.org/builders/arm64/builds/1371), [linux-dev](http://build.servo.org/builders/linux-dev/builds/1394), [linux-rel](http://build.servo.org/builders/linux-rel/builds/1456), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/1437), [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/1421), [mac-rel-wpt](http://build.servo.org/builders/mac-rel-wpt/builds/1473), [windows](http://build.servo.org/builders/windows/builds/1103)
